### PR TITLE
feat(ui5-*): integrate element internals in input components

### PR DESCRIPTION
**Pull request by nnaydenow**
Integrate the ElementInternals feature into UI5 web components. This enhancement allows input-like web components to behave as native input elements within HTML forms. As a result, the `features/InputElementsFormSupport.js` feature is now redundant and has been removed.

Changes in `@customElement` decorator: Added new `formAssociated` flag attaches internals to the component when the component is associated with form (placed inside form).

BREAKING CHANGE: The `features/InputElementsFormSupport.js` feature has been deleted and now, form elements works natively in form elements.

Related: #8461

### CODE REVIEW

File: packages/base/src/features/InputElementsFormSupport.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+export {
+	attachFormElementInternals,
+	setFormValue,
+	setFormValidity,
+	submitForm,
+	resetForm,
+};
+
+export {
```
Comment (nnaydenow) [inline packages/base/src/features/InputElementsFormSupport.ts:78]: export type

File: packages/main/src/Button.ts [hunk 2]
```diff
@@ -364,12 +354,11 @@ class Button extends UI5Element implements IFormElement, IButton {
 		}
 
 		markEvent(e, "button");
-		const formSupport = getFeature<typeof FormSupport>("FormSupport");
-		if (formSupport && this._isSubmit) {
-			formSupport.triggerFormSubmit(this);
+		if (this.internals_?.form && this._isSubmit) {
```
Comment (ilhan007) [inline packages/main/src/Button.ts:357]: submitForm and resetForm internally check for  this.internals_?.form, we can simplify the condition a little bit.

File: packages/base/src/features/InputElementsFormSupport.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+		const focusRef = await element.formElementAnchor?.();
+		element.internals_.setValidity(element.formValidity, element.formValidityMessage, focusRef);
+	} else {
+		element.internals_.setValidity({});
+	}
+};
+
+const submitForm = (element: IFormElement) => {
+	if (!element.internals_?.form) {
+		return;
```
Comment (ilhan007) [inline packages/base/src/features/InputElementsFormSupport.ts:52]: As discussed, you can shorten to single line:
element.internals_?.form.?

**Review by ilhan007 — COMMENTED** [review]
Overall, looks good to me.

We have to only commonly agree on the `updatesFormValue` new metadata entity name.

**Comment by nnaydenow** [comment]
> Overall, looks good to me.
> 
> We have to only commonly agree on the `updatesFormValue` new metadata entity name.

As discussed offline the metadata name is changed to formValue

File: packages/main/src/MultiComboBox.ts [hunk 4]
```diff
@@ -456,9 +469,42 @@ class MultiComboBox extends UI5Element {
...
+
+		if (selectedItems.length) {
+			const formData = new FormData();
+
+			for (let i = 0; i < selectedItems.length; i++) {
+				formData.append(this.name, selectedItems[i].text);
+			}
+
+			if (this.value) {
+				formData.append(this.name, this.value);
```
Comment (pskelin) [inline packages/main/src/MultiComboBox.ts:499]: value should be first, easier to check for apps. also we should document that the first value is what the user typed and the rest (if any) is the selected

File: packages/base/src/features/InputElementsFormSupport.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+const submitForm = (element: IFormElement) => {
+	element._internals?.form?.requestSubmit();
+};
+
+const resetForm = (element: IFormElement) => {
+	element._internals?.form?.reset();
+};
+
+const isInputElement = (element: IFormInputElement | IFormElement): element is IFormInputElement => {
+	return true;
```
Comment (pskelin) [inline packages/base/src/features/InputElementsFormSupport.ts:59]: this check is not doing anything, you can use a cast where it is used

File: packages/main/src/MultiInput.ts [hunk 6]
```diff
@@ -115,12 +117,38 @@ class MultiInput extends Input {
...
+
+		if (tokens.length) {
+			const formData = new FormData();
+
+			for (let i = 0; i < tokens.length; i++) {
+				formData.append(this.name, tokens[i].text);
+			}
+
+			if (this.value) {
+				formData.append(this.name, this.value);
```
Comment (pskelin) [inline packages/main/src/MultiInput.ts:143]: same as combo box, user typed value should be first (event if empty) and it should be documented

**Review by pskelin — CHANGES_REQUESTED** [review]

**Review by ilhan007 — APPROVED** [review]

**Review by pskelin — APPROVED** [review]

**Pull request merged**